subprojects/testing-jvm/src/integTest/groovy/org/gradle/testing/TestTaskSnapshottingConfigurationIntegrationTest.groovy

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.testing
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import org.gradle.test.fixtures.file.TestFile
+
+class TestTaskSnapshottingConfigurationIntegrationTest extends AbstractIntegrationSpec {
```

Comment:
> changing the resource's content, removing the resource, or adding the resource will be ignored.

Yep, we should add this.

> you cannot ignore a class.

I think we need to be able to ignore classes. For example, `GradleVersion.class` could have the version number embedded as a constant (as opposed to read from a `.properties` file).

> you can ignore a resource in a jar or a class directory.

Yep.

> ideally that you can ignore a resource from a specific library (as opposed to ignoring it from every library).

I'm not sure how you'd identify a library in this context: we have a bunch of `File` objects to process (pointing to JAR files and class directories). The only way we could identify a library would be to expect the JAR file's name to match some pattern maybe, but that sounds problematic. I think it's safe not to do this now, and come back to this feature later if users need it.